src/tools/Circle.ts

```diff
@@ -75,23 +91,52 @@ export default class Circle extends ToolClass implements CircleInterface {
   mouseUpAction = (event: CustomEvent): void => {
     if (this.enabled) {
       this.disableTool()
-      if (!this.canvasEntity.hasMoved) {
+      if (!this.hasMoved) {
         this.circleCreator.destroy()
       } else {
-        if (this.canvasElement.tool.temporary) {
+        if (this.temporary) {
           this.circleCreator.runTemporaryAnimation(this.circleCreator.getGroup())
         }
-        this.sendAndAddToState({
-          id: uuid(),
-          jti: this.canvasElement.jti,
-          modifyData: {
-            additions: [this.canvasElement.id],
-            tool: AdditionTools.CIRCLE
-          },
-          modifyType: Tracker.ADDITION,
-          canvasElements: [this.canvasElement],
-          timestampModified: ISO.timestamp()
-        }, SocketCanvasToolsEmit.CANVAS_TOOLS_CIRCLE)
+        if (this.jti) {
```

Comment:
Reason this code is longer is because we build the canvas element object in the class now instead of doing that in the canvasEntity. Its actually better this way because the canvasEntity module was very generic for all tools, but now different tools can build their canvas element exactly the way they want to; which is good